python/rascaline/rascaline/utils/clebsch_gordan/_utils.py

```diff
@@ -144,62 +143,28 @@ def parse_selected_keys(
     return selected_keys_
 
 
-def parse_bool_iteration_filters(
-    n_iterations: int,
-    skip_redundant: Union[bool, List[bool]] = False,
-    output_selection: Optional[Union[bool, List[bool]]] = None,
-    keep_l_in_keys: Optional[Union[bool, List[bool]]] = None,
-) -> Tuple[List[bool], List[bool]]:
+def parse_bool_iteration_filter(
+    n_iterations: int, bool_filter: Union[bool, List[bool]], filter_name: str
+) -> List[bool]:
     """
-    Parses the ``skip_redundant`` and ``output_selection`` arguments passed to
-    public functions.
+    Parses the ``bool_filter`` argument passed to public functions.
     """
-    if isinstance(skip_redundant, bool):
-        skip_redundant_ = [skip_redundant] * n_iterations
-    else:
-        skip_redundant_ = skip_redundant
-
-    if not all([isinstance(val, bool) for val in skip_redundant_]):
-        raise TypeError("`skip_redundant` must be a `bool` or `list` of `bool`")
-    if not len(skip_redundant_) == n_iterations:
-        raise ValueError(
-            "`skip_redundant` must be a bool or `list` of `bool` of length"
-            " `correlation_order` - 1"
-        )
-    if output_selection is None:
-        output_selection = [False] * (n_iterations - 1) + [True]
-    else:
-        if isinstance(output_selection, bool):
-            output_selection = [output_selection] * n_iterations
-        if not isinstance(output_selection, list):
-            raise TypeError("`output_selection` must be passed as `list` of `bool`")
-
-    if not len(output_selection) == n_iterations:
-        raise ValueError(
-            "`output_selection` must be a ``list`` of ``bool`` of length"
-            " corresponding to the number of CG iterations"
-        )
-    if not all([isinstance(v, bool) for v in output_selection]):
-        raise TypeError("`output_selection` must be passed as a `list` of `bool`")
-    if not all([isinstance(v, bool) for v in output_selection]):
-        raise TypeError("`output_selection` must be passed as a `list` of `bool`")
-
-    if isinstance(keep_l_in_keys, bool):
-        keep_l_in_keys_ = [
-            keep_l_in_keys if output else False for output in output_selection
-        ]
+    if bool_filter is None:
+        bool_filter = [False] * (n_iterations - 1) + [True]
```

Comment:
this is only intended for output selection, right? I would then handle it outside of this function for better clarity

Reply:
Yes true

Reply:
done!